optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -52,11 +53,46 @@
 
     private static final AlphabeticMemberComparator alphabeticMemberComparator = new AlphabeticMemberComparator();
 
-    public static <T> T newInstance(Object bean, String propertyName, Class<T> clazz) {
+    /**
+     * Create a new instance of clazz from a config's property.
+     * <p>
+     * If the instantiation fails, the simple class name of {@code configBean} will be used as the owner of
+     * {@code propertyName}.
+     * <p>
+     * Intended usage:
+     *
+     * <pre>
+     * selectionFilter = ConfigUtils.newInstance(
+     *         config, "filterClass", config.getFilterClass());
+     * </pre>
+     *
+     * @param configBean the bean holding the {@code clazz} to be instantiated
+     * @param propertyName {@code configBean}'s property holding {@code clazz}
+     * @param clazz {@code Class} representation of the type {@code T}
+     * @param <T> the new instance type
+     * @return new instance of clazz
+     */
+    public static <T> T newInstance(Object configBean, String propertyName, Class<T> clazz) {
+        return ConfigUtils.newInstance(() -> configBean.getClass().getSimpleName(), propertyName, clazz);
+    }
+
+    /**
+     * Create a new instance of clazz from a general source.
+     * <p>
+     * If the instantiation fails, the result of {@code ownerDescriptor} will be used to describe the owner of
+     * {@code propertyName}.
+     *
+     * @param ownerDescriptor describes the owner of {@code propertyName}
+     * @param propertyName property holding the {@code clazz}
+     * @param clazz {@code Class} representation of the type {@code T}
+     * @param <T> the new instance type
+     * @return new instance of clazz
+     */
+    public static <T> T newInstance(Supplier<String> ownerDescriptor, String propertyName, Class<T> clazz) {
         try {
             return clazz.newInstance();
```

Comment:
@yurloc Please address this deprecation, since you're already touching this code.